### PR TITLE
#1722 - ignore apps with zero apps

### PR DESCRIPTION
**Pull request by rohit01**

### CODE REVIEW

File: retrieval/discovery/marathon/marathon_test.go [hunk 1]
```diff
@@ -92,6 +91,52 @@ func marathonTestAppList(labels map[string]string, runningTasks int) *AppList {
...
+		case tgs := <-ch:
+			tg := tgs[0]
+
+			if tg.Source != "test-service-zero-ports" {
+				t.Fatalf("Wrong target group name: %s", tg.Source)
+			}
+			if len(tg.Targets) != 0 {
+				t.Fatalf("Wrong number of targets: %v", tg.Targets)
+			}
+		default:
```
Comment (fabxc) [inline retrieval/discovery/marathon/marathon_test.go]: This should be racy as the default case is executed even if the channel send is just a bit delayed, which depends on the runtime. This should be a `case <-time.After(5 * time.Second)`.

**Comment by fabxc** [comment]
Thanks! Can you squash those commits and fix the commit message ("zero apps" -> "zero ports").
Prefixing the commit message with `provider/marathon: ...` would be appreciated.

**Comment by rohit01** [comment]
Not sure, why it was showing 44 commits. Recreated the branch and applied the changes once again.

Here's the new PR: https://github.com/prometheus/prometheus/pull/1729

File: retrieval/discovery/marathon/marathon_test.go (hunk 1 above)
Comment (beorn7) [inline retrieval/discovery/marathon/marathon_test.go]: So this seems to be a problem in all the tests, not just the new one. Will create a PR.

Comment (fabxc) [inline retrieval/discovery/marathon/marathon_test.go]: #1733

**Pull request closed without merging**